catalogue/webapp/test/components/ItemRequestButton.test.js

```diff
@@ -0,0 +1,108 @@
+// @flow
+import {
+  mountWithTheme,
+  updateWrapperAsync,
+} from '@weco/common/test/fixtures/enzyme-helpers';
+import ItemRequestButton from '@weco/catalogue/components/ItemRequestButton/ItemRequestButton';
+import useAuth from '@weco/common/hooks/useAuth';
+import { getUserHolds } from '@weco/catalogue/services/stacks/requests';
+import mockAuthStates from '@weco/catalogue/__mocks__/auth-states';
+import mockWork from '@weco/catalogue/__mocks__/stacks-work';
+import mockRequests from '@weco/catalogue/__mocks__/stacks-requests';
+
+const availableItem = mockWork.items.find(
+  item => item.status.label === 'Available'
+);
+const unavailableItem = mockWork.items.find(
+  item => item.status.label !== 'Available'
+);
+const unrequestableItem = mockWork.items.find(
+  item => item.status.label === 'Missing'
+);
+
+jest.mock('@weco/catalogue/services/stacks/requests', () => ({
+  getUserHolds: jest.fn(),
+}));
+jest.mock('@weco/common/hooks/useAuth');
+
+describe('Feature: 2. As a library member I want to request an item', () => {
+  (getUserHolds: any).mockResolvedValue(mockRequests);
+  test(`Scenario: I'm logged out, viewing a work page
+  Given the work has a requestable item
+  And it’s available to request
+  Then I can see a primary CTA to log in`, async () => {
+    (useAuth: any).mockImplementation(() => {
+      return mockAuthStates.unauthorized;
+    });
```

Comment:
Slightly more readable?
```suggestion
    (useAuth: any).mockImplementation(() => mockAuthStates.unauthorized);
```